src/libsyntax/parse/parser.rs

```diff
@@ -1087,6 +1087,11 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Returns whether any of the given keywords are `dist` tokens ahead of the current one.
+    fn is_keyword_ahead(&self, dist: usize, kws: &[Symbol]) -> bool {
```

Comment:
We could possibly reduce this to just one keyword... that is one option... :)

Reply:
18 single keyword vs 5 multiple keywords.
\+ `look_ahead` is a single array lookup and a match, more or less, so not much reason to optimize.
Might be reasonable to reduce to one keyword.